chaincode/academy/go/certificate.go

```diff
@@ -1755,6 +1757,40 @@ func QueryClassesOfStudent(stub shim.ChaincodeStubInterface, args []string) sc.R
 	return shim.Success(jsonRow)
 }
 
+func QueryCoursesOfStudent(stub shim.ChaincodeStubInterface, args []string) sc.Response {
+	if len(args) != 1 {
+		return shim.Error("Incorrect number of arguments. Expecting 1")
+	}
+
+	StudentUsername := args[0]
+
+	student, err := getStudent(stub, "Student-" + StudentUsername)
+
+	if err != nil {
+		return shim.Error("Student dose not exist - " + StudentUsername)
+	}
+
+	var tlist []Course
+	var i int
+
+	for i = 0; i < len(student.Courses); i++ {
+
+		course, err := getCourse(stub, "Course-" + student.Courses[i])
+		if err != nil {
+			return shim.Error("Course does not exist - " + student.Courses[i])
+		}
+		tlist = append(tlist, course)
+	}
+
+	jsonRow, err := json.Marshal(tlist)
+
+	if err != nil {
+		return shim.Error("JSON Marshal has failed")
```

Comment:
để cái này vẫn khó hiểu, vì nó là một lỗi rất chung chung
e nên để nội dung nó liên quan đến function mình đang làm ý, ví dụ `Cannot json encode list courses of student` chẳng hạn, để sau này có debug thì cũng dễ dàng hơn